pkg/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -118,11 +95,31 @@ func (c *collector) extractBOMWithTrivy(ctx context.Context, storedImage *worklo
 
 			// Updating workloadmeta entities directly is not thread-safe, that's why we
 			// generate an update event here instead.
-			if err := c.handleImageCreateOrUpdate(ctx, storedImage.Namespace, storedImage.Name, &sbom); err != nil {
-				log.Warnf("Error extracting SBOM for image: namespace=%s name=%s, err: %s", storedImage.Namespace, storedImage.Name, err)
+			if err := c.handleImageCreateOrUpdate(ctx, result.ImgMeta.Namespace, result.ImgMeta.Name, &sbom); err != nil {
+				log.Warnf("Error extracting SBOM for image: namespace=%s name=%s, err: %s", result.ImgMeta.Namespace, result.ImgMeta.Name, err)
 			}
 		}
 	}()
 
 	return nil
 }
+
+func (c *collector) extractBOMWithTrivy(ctx context.Context, storedImage *workloadmeta.ContainerImageMetadata, resultChan chan<- sbom.ScanResult) error {
```

Comment:
it look like a typo in the function  name

```suggestion
func (c *collector) extractSBOMWithTrivy(ctx context.Context, storedImage *workloadmeta.ContainerImageMetadata, resultChan chan<- sbom.ScanResult) error {
```

Reply:
since it is a private method, maybe you can fixe the typo

Reply:
I did it so it should be good